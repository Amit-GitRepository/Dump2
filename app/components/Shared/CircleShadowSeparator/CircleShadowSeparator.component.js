import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './CircleShadowSeparator.styles';
import {ISText} from '../../Shared';
import {View} from 'react-native';

class CircleShadowSeparator extends Component {
  render () {
    return (
      <View style={styles.container}>
        <View style={styles.topHalf} />
        <View style={styles.bottomHalf} />
        <View style={styles.semiCircle}><ISText type='BOLD' style={styles.text}>{this.props.text}</ISText></View>
      </View>
    );
  }
}

CircleShadowSeparator.defaultProps = {
  text: ''
};

CircleShadowSeparator.propTypes = {
  text: PropTypes.string
};

export default CircleShadowSeparator;
