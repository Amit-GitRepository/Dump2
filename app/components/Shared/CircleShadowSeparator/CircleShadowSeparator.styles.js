import {colors, fonts} from '../../../themes/constants.styles';

export default {
  container: {
    alignSelf: 'stretch',
    justifyContent: 'center',
    flexDirection: 'column'
  },
  semiCircle: {
    position: 'absolute',
    top: 10,
    alignSelf: 'center',
    backgroundColor: colors.PRIMARY_BG_TEXT_CONTRAST,
    paddingHorizontal: 10,
    paddingBottom: 4,
    borderBottomLeftRadius: 50,
    borderBottomRightRadius: 50
  },
  text: {
    color: colors.LANDING_TITLE_BG,
    fontSize: fonts.FONT_SIZE_SMALL,
    backgroundColor: colors.TRANSPARENT
  },
  topHalf: {
    flex: 1,
    padding: 10,
    elevation: 1
  },
  bottomHalf: {
    flex: 1,
    padding: 10,
    backgroundColor: colors.PRIMARY_BG_SEPARATOR
  }
};
