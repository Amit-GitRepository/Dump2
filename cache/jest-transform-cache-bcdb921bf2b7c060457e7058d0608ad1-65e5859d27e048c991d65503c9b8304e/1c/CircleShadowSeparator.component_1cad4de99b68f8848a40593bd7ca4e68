Object.defineProperty(exports,"__esModule",{value:true});var cov_1eboh6lcod=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/CircleShadowSeparator/CircleShadowSeparator.component.js',hash='ce60ccf5d14125165c0613aacaf0de859aafa8fd',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/CircleShadowSeparator/CircleShadowSeparator.component.js',statementMap:{'0':{start:{line:10,column:42},end:{line:10,column:52}},'1':{start:{line:11,column:26},end:{line:11,column:53}},'2':{start:{line:12,column:29},end:{line:12,column:59}},'3':{start:{line:13,column:4},end:{line:19,column:6}},'4':{start:{line:23,column:0},end:{line:27,column:2}},'5':{start:{line:29,column:0},end:{line:33,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:9,column:2},end:{line:9,column:3}},loc:{start:{line:9,column:12},end:{line:20,column:3}},line:9}},branchMap:{},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0},f:{'0':0},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _CircleShadowSeparator=require('./CircleShadowSeparator.styles');var _CircleShadowSeparator2=_interopRequireDefault(_CircleShadowSeparator);
var _constants=require('../../../themes/constants.styles');
var _Shared=require('../../Shared');
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

CircleShadowSeparator=function(_Component){_inherits(CircleShadowSeparator,_Component);function CircleShadowSeparator(){_classCallCheck(this,CircleShadowSeparator);return _possibleConstructorReturn(this,(CircleShadowSeparator.__proto__||Object.getPrototypeOf(CircleShadowSeparator)).apply(this,arguments));}_createClass(CircleShadowSeparator,[{key:'render',value:function render()
{cov_1eboh6lcod.f[0]++;var _ref=(cov_1eboh6lcod.s[0]++,
this.props),text=_ref.text,topColor=_ref.topColor,bottomColor=_ref.bottomColor;
var topColorStyle=(cov_1eboh6lcod.s[1]++,{backgroundColor:topColor});
var bottomColorStyle=(cov_1eboh6lcod.s[2]++,{backgroundColor:bottomColor});cov_1eboh6lcod.s[3]++;
return(
_react2.default.createElement(_reactNative.View,{style:_CircleShadowSeparator2.default.container},
_react2.default.createElement(_reactNative.View,{style:[_CircleShadowSeparator2.default.topHalf,topColorStyle]}),
_react2.default.createElement(_reactNative.View,{style:[_CircleShadowSeparator2.default.bottomHalf,bottomColorStyle]}),
_react2.default.createElement(_reactNative.View,{style:_CircleShadowSeparator2.default.semiCircle},_react2.default.createElement(_Shared.ISText,{type:'BOLD',style:_CircleShadowSeparator2.default.text},text))));


}}]);return CircleShadowSeparator;}(_react.Component);cov_1eboh6lcod.s[4]++;


CircleShadowSeparator.defaultProps={
text:'',
topColor:_constants.colors.GRAY_NURSE,
bottomColor:_constants.colors.PRIMARY_BG_TEXT_CONTRAST};cov_1eboh6lcod.s[5]++;


CircleShadowSeparator.propTypes={
text:_propTypes2.default.string,
topColor:_propTypes2.default.string,
bottomColor:_propTypes2.default.string};exports.default=


CircleShadowSeparator;