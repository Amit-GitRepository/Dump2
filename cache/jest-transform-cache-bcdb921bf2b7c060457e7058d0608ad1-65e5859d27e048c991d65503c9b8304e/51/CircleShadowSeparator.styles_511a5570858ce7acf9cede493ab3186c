Object.defineProperty(exports,"__esModule",{value:true});var cov_1rbvufbv1c=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/CircleShadowSeparator/CircleShadowSeparator.styles.js',hash='bedfcfee1c8d4b790900db1541e6c2d93d5fc773',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/CircleShadowSeparator/CircleShadowSeparator.styles.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _constants=require('../../../themes/constants.styles');exports.default=

{
container:{
alignSelf:'stretch',
justifyContent:'center',
flexDirection:'column'},

semiCircle:{
position:'absolute',
top:10,
alignSelf:'center',
backgroundColor:_constants.colors.PRIMARY_BG_SEPARATOR,
paddingHorizontal:10,
paddingBottom:4,
borderBottomLeftRadius:50,
borderBottomRightRadius:50,
shadowColor:_constants.colors.SHADOW,
shadowOffset:{width:0,height:1},
shadowOpacity:0.6,
shadowRadius:1,
elevation:1},

text:{
color:_constants.colors.LANDING_TITLE_BG,
fontSize:_constants.fonts.FONT_SIZE_SMALL,
backgroundColor:_constants.colors.TRANSPARENT},

topHalf:{
flex:1,
padding:10,
elevation:1},

bottomHalf:{
flex:1,
padding:10,
shadowColor:_constants.colors.SHADOW,
shadowOffset:{width:0,height:0},
shadowOpacity:0.6,
shadowRadius:3}};