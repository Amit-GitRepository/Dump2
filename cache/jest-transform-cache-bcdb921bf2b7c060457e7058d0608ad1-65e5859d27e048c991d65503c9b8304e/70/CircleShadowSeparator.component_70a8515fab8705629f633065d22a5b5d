Object.defineProperty(exports,"__esModule",{value:true});var cov_1eboh6lcod=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/CircleShadowSeparator/CircleShadowSeparator.component.js',hash='5e9d149d1e233d1b246f0533cdfb13b863d794cb',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/CircleShadowSeparator/CircleShadowSeparator.component.js',statementMap:{'0':{start:{line:9,column:4},end:{line:15,column:6}},'1':{start:{line:19,column:0},end:{line:21,column:2}},'2':{start:{line:23,column:0},end:{line:25,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:8,column:2},end:{line:8,column:3}},loc:{start:{line:8,column:12},end:{line:16,column:3}},line:8}},branchMap:{},s:{'0':0,'1':0,'2':0},f:{'0':0},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _CircleShadowSeparator=require('./CircleShadowSeparator.styles');var _CircleShadowSeparator2=_interopRequireDefault(_CircleShadowSeparator);
var _Shared=require('../../Shared');
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

CircleShadowSeparator=function(_Component){_inherits(CircleShadowSeparator,_Component);function CircleShadowSeparator(){_classCallCheck(this,CircleShadowSeparator);return _possibleConstructorReturn(this,(CircleShadowSeparator.__proto__||Object.getPrototypeOf(CircleShadowSeparator)).apply(this,arguments));}_createClass(CircleShadowSeparator,[{key:'render',value:function render()
{cov_1eboh6lcod.f[0]++;cov_1eboh6lcod.s[0]++;
return(
_react2.default.createElement(_reactNative.View,{style:_CircleShadowSeparator2.default.container},
_react2.default.createElement(_reactNative.View,{style:_CircleShadowSeparator2.default.topHalf}),
_react2.default.createElement(_reactNative.View,{style:_CircleShadowSeparator2.default.bottomHalf}),
_react2.default.createElement(_reactNative.View,{style:_CircleShadowSeparator2.default.semiCircle},_react2.default.createElement(_Shared.ISText,{type:'BOLD',style:_CircleShadowSeparator2.default.text},this.props.text))));


}}]);return CircleShadowSeparator;}(_react.Component);cov_1eboh6lcod.s[1]++;


CircleShadowSeparator.defaultProps={
text:''};cov_1eboh6lcod.s[2]++;


CircleShadowSeparator.propTypes={
text:_propTypes2.default.string};exports.default=


CircleShadowSeparator;