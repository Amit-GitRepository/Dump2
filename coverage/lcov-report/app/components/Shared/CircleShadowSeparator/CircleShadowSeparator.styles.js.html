<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/components/Shared/CircleShadowSeparator/CircleShadowSeparator.styles.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">app/components/Shared/CircleShadowSeparator</a> CircleShadowSeparator.styles.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/0</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import {colors, fonts} from '../../../themes/constants.styles';
&nbsp;
export default {
  container: {
    alignSelf: 'stretch',
    justifyContent: 'center',
    flexDirection: 'column'
  },
  semiCircle: {
    position: 'absolute',
    top: 10,
    alignSelf: 'center',
    backgroundColor: colors.PRIMARY_BG_SEPARATOR,
    paddingHorizontal: 10,
    paddingBottom: 4,
    borderBottomLeftRadius: 50,
    borderBottomRightRadius: 50,
    shadowColor: colors.SHADOW,
    shadowOffset: {width: 0, height: 1},
    shadowOpacity: 0.6,
    shadowRadius: 1,
    elevation: 1
  },
  text: {
    color: colors.LANDING_TITLE_BG,
    fontSize: fonts.FONT_SIZE_SMALL,
    backgroundColor: colors.TRANSPARENT
  },
  topHalf: {
    flex: 1,
    padding: 10,
    elevation: 1
  },
  bottomHalf: {
    flex: 1,
    padding: 10,
    shadowColor: colors.SHADOW,
    shadowOffset: {width: 0, height: 0},
    shadowOpacity: 0.6,
    shadowRadius: 3
  }
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Oct 31 2017 11:24:42 GMT+0530 (IST)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
