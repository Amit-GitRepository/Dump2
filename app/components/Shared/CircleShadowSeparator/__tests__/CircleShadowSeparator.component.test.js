import CircleShadowSeparator from '../CircleShadowSeparator.component';
import React from 'react';
import ReactSixteenAdapter from 'enzyme/build/adapters/ReactSixteenAdapter';
import renderer from 'react-test-renderer'; // Enzyme adapter for React 16
import {configure, shallow} from 'enzyme';

configure({adapter: new ReactSixteenAdapter()});

describe('CircleShadowSeparator component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<CircleShadowSeparator />);
    const component = renderer.create(<CircleShadowSeparator />).toJSON();
    expect(component).toMatchSnapshot();
    expect(wrapper).toBeDefined();
  });
  it('renders correctly', () => {
    const wrapper = shallow(<CircleShadowSeparator/>);
    expect(wrapper).toBeDefined();
  });
});
