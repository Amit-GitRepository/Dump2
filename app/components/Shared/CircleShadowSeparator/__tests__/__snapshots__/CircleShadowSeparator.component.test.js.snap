// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleShadowSeparator component renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "elevation": 1,
        "flex": 1,
        "padding": 10,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#E9E9EA",
        "flex": 1,
        "padding": 10,
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomLeftRadius": 50,
        "borderBottomRightRadius": 50,
        "paddingBottom": 4,
        "paddingHorizontal": 10,
        "position": "absolute",
        "top": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#605E71",
          "fontFamily": "SukhumvitSet-Bold",
          "fontSize": 14,
        }
      }
    >
      
    </Text>
  </View>
</View>
`;
